File: Project Οντοκεντρικός Προγραμματισμός/C++ Project/CodeBlocks Project Hotel/src/RoomTypeE.cpp
#include <RoomTypeE.h>
#include <Booking.h>

RoomTypeE::RoomTypeE(int minPeople, int minDays,int maxPeople,int pricePerPerson)//Constructor for a RoomTypeE type object.
{
	setMinPeople(minPeople);
	setMinDays(minDays);
	setMaxPeople(maxPeople);
	setPricePerPerson(pricePerPerson);
}

RoomTypeE::~RoomTypeE()//Default destructor for a RoomTypeE type object.
{
}

double RoomTypeE::getPriceOfRoom()//Function that returns the price of a room type E.
{
	int i = 0;
	double moneyEarnedFromRoom = 0;
	double discountPerPerson;
	for (i = 1; i<31; i++)
	{
		if (roomAvailability[i] != 0)
		{
			if (i <= 3)
			{
				discountPerPerson = (this->getPricePerPerson() / 2);//Calculating the discount per day for a room type E.
				moneyEarnedFromRoom += (roomAvailability[i]->getPeopleNum()*(this->getPricePerPerson() - discountPerPerson));
				continue;
			}
			moneyEarnedFromRoom += (roomAvailability[i]->getPeopleNum() * this->getPricePerPerson());
		}
	}
	return moneyEarnedFromRoom;
}

File: Project Οντοκεντρικός Προγραμματισμός/Java Project/Hotel Project, Swing/RoomTypeE.java
public class RoomTypeE extends RoomTypeC//Δωμάτιο τύπου C,αλλά με έκπτωση 50% ανά άτομο μετά την τρίτη μέρα.
{
     public double getPriceOfRoom()//Μέθοδος για τον υπολογισμό των εσόδων δωματίου τύπου E,λαμβάνοντας υπόψη τα ελάχιστα//
    {                              //άτομα και μέρες,καθώς και την έκπτωση.
        int i=0;
        double moneyEarnedFromRoom=0; //Μεταβλητή για έσοδα από δωμάτιο.
        double discountPerPerson; //Μεταβλητή για έκπτωση ανά άτομο.
        for(i=1; i<31; i++)
        {
            if(roomAvailability[i]!=null)
                {
                    if(i<=3)
                    {
                        discountPerPerson = (getPricePerPerson() / 2 );
                        moneyEarnedFromRoom+=(roomAvailability[i].getPeopleNum()*( getPricePerPerson() - discountPerPerson));
                        continue;
                    }
                    moneyEarnedFromRoom = (moneyEarnedFromRoom + (roomAvailability[i].getPeopleNum() * getPricePerPerson()));
                }
        }
        return moneyEarnedFromRoom;
    }



}
